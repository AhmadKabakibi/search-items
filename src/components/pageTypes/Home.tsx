import React from 'react'
import Helmet from 'react-helmet'
import styled, { keyframes } from 'styled-components'
import colors from '../../styles/colors'
import textStyles from '../../styles/textStyles'

// Components
import Button from '../Button'
import Header from '../Header'

import logo from './logo.svg'

const Home = () => (
  <Wrapper>
    <Helmet title="Home" />
    <Header title="Header" color={colors.creamy} />
    <App>
      <AppHeader>
        <AppLogo src={logo} alt="logo" />
        <Copy>React App with TypeScript</Copy>
        <Button
          onClick={() => {
            alert('Ochh youuu :P ')
          }}
        >
          Do action
        </Button>
      </AppHeader>
    </App>
  </Wrapper>
)

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
`

const App = styled.div`
  text-align: center;
`
const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

const AppLogo = styled.img`
  animation: ${rotate} infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
`
const Copy = styled.p`
  ${textStyles.title};
  color: ${colors.black};
`
const AppHeader = styled.header`
  flex-direction: column;
  align-items: center;
  justify-content: center;
  display: flex;
  min-height: 100vh;
  background-color: ${colors.white};
  color: white;
  font-size: 1.4em;
`
export default Home
