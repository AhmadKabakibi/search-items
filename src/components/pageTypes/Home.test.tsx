import 'jest-styled-components'
import React from 'react'

// Components
import Home from './Home'

const Component = shallow(<Home />)

describe('Home', () => {
  it('should render a Header', () => {
    expect(Component.find('Header')).toHaveLength(1)
  })
})

describe('Home:Button', () => {
  const Button = Component.find('Button')

  it('should render an action button', () => {
    expect(Button).toHaveLength(1)
  })
})
