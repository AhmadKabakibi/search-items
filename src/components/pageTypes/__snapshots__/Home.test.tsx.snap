// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render a Home 1`] = `
<Home__Wrapper>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Home"
  />
  <Header
    color="#EFF0EA"
    title="Multiple Select Filter"
  />
  <Home__Container>
    <MultiSelectFilter
      data={
        Array [
          "foo",
          "bar",
        ]
      }
    />
  </Home__Container>
</Home__Wrapper>
`;
